File: LeetCode/GoLang/GroupAnagrams49.go
package main

import (
	"fmt"
	"sort"
	"strings"
)

// 排序解法
func main() {
	question := [6]string{"eat", "tea", "tan", "ate", "nat", "bat"}

	// 申请一个map
	strMapping := make(map[string][]string, 10)

	// 循环处理
	for _, value := range question {
		key := sortStrings(value)
		if _, ok := strMapping[key]; !ok {
			strMapping[key] = []string{}
		}
		strMapping[key] = append(strMapping[key], value)
	}

	// 将答案塞入到最后答案数组中
	var ans [][]string
	for _, value := range strMapping {
		ans = append(ans, value)
	}

	fmt.Println(strMapping, ans)
}

// 给字符串排序
func sortStrings(str string) string {
	// 切分字符串
	split := strings.Split(str, "")

	// 排序
	sort.Strings(split)

	// 组装string
	return strings.Join(split, "")
}

File: LeetCode/Algorithms/GroupAnagrams49.php
<?php

## 解法1：排序 https://leetcode.cn/problems/group-anagrams/

$question = ["eat", "tea", "tan", "ate", "nat", "bat"];

$answer = [];
foreach ($question as $value) {
    $sortStr = sortStr($value);

    $answer[$sortStr][] = $value;
}

/**
 * 字符串排序
 *
 * @param string $waitString 等待排序的字符串
 * @return string
 */
function sortStr(string $waitString): string
{
    $strArr = str_split($waitString);

    asort($strArr);

    return implode("", $strArr);
}

var_dump($answer);